### PR TITLE
fix(condo): DOMA-9373 fix archived and paid receipts, tests, commented sendResidentsNoAccountNotifications

### PR DESCRIPTION
**Pull request by vovaaxeapolla**

### CODE REVIEW

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/resident/tasks/helpers/sendBillingReceiptNotifications.js [hunk 1]
```diff
@@ -26,8 +26,8 @@ const sendBillingReceiptNotifications = async (context = null) => {
 
         return { status: DISABLED }
     }
-
-    await sendResidentsNoAccountNotifications()
+    //TODO: DOMA-10913 This func needs to be refactored and optimized or removed, currently it falls by time out
+    // await sendResidentsNoAccountNotifications()
```
Comment (AleX83Xpert) [inline apps/condo/domains/resident/tasks/helpers/sendBillingReceiptNotifications.js:30]: This function fails for organizations with huge amount of BillingProperties. But still works for small organizations.

**Review by AleX83Xpert — APPROVED** [review]

File: apps/condo/domains/resident/tasks/sendBillingReceiptsAddedNotificationForOrganizationContextTask.js [hunk 2]
```diff
@@ -175,9 +193,9 @@ async function prepareReceiptsData (receipts, context) {
 }
 
 async function processReceiptData (receipt, context) {
-    const account = await getById('BillingAccount', receipt.account)
+    const account = await getById('BillingAccount', receipt.account.id)
```
Comment (YEgorLu) [inline apps/condo/domains/resident/tasks/sendBillingReceiptsAddedNotificationForOrganizationContextTask.js:196]: What about "deletedAt"? Maybe `getByCondition('BillingAccount', { id: receipt.account.id, deletedAt: null })`?

File: apps/condo/domains/resident/tasks/sendBillingReceiptsAddedNotificationForOrganizationContextTask.js [hunk 3]
```diff
@@ -18,8 +21,11 @@ const {
 } = require('@condo/domains/notification/constants/constants')
 const { sendMessage } = require('@condo/domains/notification/utils/serverSchema')
 const { BILLING_CONTEXT_SYNCHRONIZATION_DATE } = require('@condo/domains/resident/constants/constants')
-const logger = getLogger('sendNewBillingReceiptNotification')
 
+const logger = getLogger('sendNewBillingReceiptNotification')
+const BILLING_RECEIPT_FIELDS = 'id period toPay toPayDetails { charge } createdAt isPayable '
+    + 'context { id organization { id } integration { id currencyCode } } '
+    + 'account { id number } property { id address } category { id  nameNonLocalized }'
```
Comment (YEgorLu) [inline apps/condo/domains/resident/tasks/sendBillingReceiptsAddedNotificationForOrganizationContextTask.js:28]: If I understood correctly, you don't use all these fields. You are using only  id, period, toPay, toPayDetails, and re-requesting other fields

So better remove unused relations

File: apps/condo/domains/resident/tasks/sendBillingReceiptsAddedNotificationForOrganizationContextTask.js [hunk 4]
```diff
@@ -144,18 +152,28 @@ async function sendBillingReceiptsAddedNotificationForOrganizationContext (conte
...
+    const { keystone: context } = getSchemaCtx('BillingReceipt')
     const receiptsWhere = {
         createdAt_gt: receiptCreatedAfter,
         context: { id: contextId },
-        period_in: periods,
         toPay_gt: '0',
         deletedAt: null,
     }
-    return find('BillingReceipt', receiptsWhere)
+    const receipts = await BillingReceipt.getAll(context, receiptsWhere, BILLING_RECEIPT_FIELDS)
```
Comment (YEgorLu) [inline apps/condo/domains/resident/tasks/sendBillingReceiptsAddedNotificationForOrganizationContextTask.js:162]: Will that work?
We have restriction: in one request there can be 1000 entities at max (relations interpreted as entities)

It means 1000 flat receipts, or 500 receipts with BillingIntegrationContext, 333 receipts with context and BillingProperty and so on.

Maybe try to do it chunk by chunk or using flat `find`?

**Pull request merged**